src/rust/src/backend/ciphers.rs

```diff
@@ -111,9 +116,36 @@ impl CipherContext {
         Ok(CipherContext {
             ctx,
             py_mode: mode.into(),
+            side,
+            update_nonce_allowed,
         })
     }
 
+    fn update_nonce(&mut self, nonce: CffiBuf<'_>) -> CryptographyResult<()> {
+        if !self.update_nonce_allowed {
```

Comment:
Rather than this attribute, let's just check `py_mode` with fresh isinstances here.